### ISSUE TITLE
Change in device name is propagated to all configs that use the device

### ISSUE THREAD

**Issue by JaimeLeon2** [MobiFlight/MobiFlight-Connector#869]
**Is your feature request related to a problem? Please describe.**
If a user changes a device name in a board, he now has to manually locate the one or more configs that use this device because Mobiflight is not able to locate the name of the assigned device for that input or output config. 

**Describe the solution you'd like**
When the name of an existing device is changed, MF should store the old name. When the board config change is accepted, MF should now search all input and output configs that use the old name and replace it with the new name. 

**Describe alternatives you've considered**
Manually searching and editing the configs is cumbersome. Maybe a unique device ID could be used, just like with the input/output configs, so that names are no longer the identifier used and name changes can be more easily propagated using this unique identifier. 

**Additional context**
Optionally, the Search of the old name and replacing with the new device name could be extended to all mcc files that make use of the board where the name change was made, thus maintaining the integrity of all mcc files.

**Comment by DocMoebiuz** [comment]
True. That is a nice feature. After uploading the config the UI could detect that you made changes to certain devices and then you could get asked if you wanted to replace those references.

Steps:
* Keep track of all devices that have changed including the new and the old name.
* Offer subsitution if user wants it

Further cases:
* User removes a device > potentially offer to auto-disable the config.
* User wants to initiate the rename to a later point in time, not directly after uploading > similar to the orphaned serials dialog?

**Comment by Pitj3** [comment]
Perhaps we can do a OnChangeName event that auto updates all the devices that use it with the name name

**Comment by JaimeLeon2** [comment]
The same logic applies to the board names.  MF Connector apparently uses the board name+serial number to identify a board.  Why not use only the serial, which is already a unique identifier?  Using only the serial number would allow users to rename a board and have Mobiflight use whatever name label is used, as long as the serial number matches.
Currently, attempting to change a board name will require the user to reselect the board in every input and output config that this board is used.  Totally unreasonable, IMHO.
